sql/sql_table.cc

```diff
@@ -9814,10 +9822,29 @@ bool mysql_checksum_table(THD *thd, TABLE_LIST *tables,
                   break;
                 }
                 default:
-                  row_crc= my_checksum(row_crc, f->ptr, f->pack_length());
+                  if (checksum_start)
+                  {
+                    if (checksum_start + checksum_length == f->ptr)
+                    {
+                      checksum_length+= f->pack_length();
+                    }
+                    else
+                    {
+                      row_crc= my_checksum(row_crc, checksum_start, checksum_length);
+                      checksum_start= NULL;
```

Comment:
Quite right. Thanks. It didn't seem to hit this branch in testing however as its engine specific its worth having this branch.